language-server/docs/ExtensionnServices.md

```diff
@@ -0,0 +1,160 @@
+
+<a name="ServerCapability"></a>
+### Server Capability Registration
+Once the service implementation is registered, it is important to register the server capabilities. The server capability registration allows the server to notify the clients on the operation capabilities exposed with a particular custom service.
+
+The speccific server capabilities data model represents the signature of the server capabilities of the particular service implementation. The data types of the specific operation related fields should be serializable.
+
+```Java
+public class CustomServiceServerCapabilities extends BallerinaServerCapability {
+    private boolean supportCustomOperation;
+    // getters and setters
+
+    public CustomServiceClientCapabilities() {
+        super(nameOfService)
+    }
+}
+```
+
+The sever capability setter specifies the server capability setting approach. The extension service can define how the capabilities are set by overriding the `build` method. There is another `build` method, which will take a json object as the argument, and that will not be used when creating the server extensions but will be used when generating the client extensions from java.
+
+```Java
+@JavaSPIService("org.ballerinalang.langserver.commons.registration.BallerinaServerCapabilitySetter")
+public class CustomServiceServerCapabilitySetter
+        extends BallerinaServerCapabilitySetter<CustomServiceServerCapabilities> {
+    @Override
+    public Optional<CustomServiceServerCapabilities> build() {
+        CustomServiceServerCapabilities capabilities = new CustomServiceServerCapabilities();
+        capabilities.setSupportCustomOperation(true);
+        
+        return Optional.of(capabilities);
+    }
+    
+    public String getCapabilityName() {
+        return nameOfService;
+    }
+
+    @Override
+    public Class<CustomServiceServerCapabilities> getCapability() {
+        return CustomServiceServerCapabilities.class;
+    }
+}
+```
+
+<a name="ClientCapability"></a>
+### Client Capability Registration
+Once the service implementation is registered, it is important to register the client capabilities. The client capability registration allows the client to notify the server on the operation capabilities supported by the client.
+
+The speccific client capabilities data model represents the signature of the client capabilities of the particular service implementation. The data types of the specific operation related fields should be serializable.
+
+```Java
+public class CustomServiceClientCapabilities extends BallerinaClientCapability {    
+    private boolean supportCustomOperation;
+
+    // getters and setters
+    public CustomServiceClientCapabilities() {
+        super(nameOfService);
+    }
+}
+```
+
+The client capability setter specifies the server capability setting approach. For the server extensions, there is no need to override the `build` methods, since the default behaviour would be enough.
```

Comment:
```suggestion
The client capability setter specifies the client capability setting approach. For the server extensions, there is no need to override the `build` methods, since the default behavior would be enough.
```